astropy/table/tests/test_table.py

```diff
@@ -3023,3 +3038,14 @@ def test_remove_columns_invalid_names_messages():
     with pytest.raises(KeyError,
                        match='columns {\'[de]\', \'[de]\'} do not exist'):
         t.remove_columns(['c', 'd', 'e'])
+
+
+def test_read_write_tilde_path(home_is_tmpdir):
+    test_file = os.path.join('~', 'test.csv')
```

Comment:
Need to test this for `pathlib.Path` as well. (I'm sure it will work but still...).

Reply:
Good point, and I may need to change this in the PRs for the other submodules as well